src/server/debug/server/server.go

```diff
@@ -105,39 +109,60 @@ func (s *debugServer) handleRedirect(
 							return err
 						}
 						return collectDebugStream(tw, r)
-
 					}
 					pod, err := s.env.GetKubeClient().CoreV1().Pods(s.env.Config().Namespace).Get(ctx, f.Worker.Pod, metav1.GetOptions{})
 					if err != nil {
 						return errors.EnsureStack(err)
 					}
 					return s.handleWorkerRedirect(ctx, tw, pod, collectWorker, redirect)
+				case *debug.Filter_Database:
+					if err := s.collectDatabaseStats(ctx, tw, postgresPrefix); err != nil {
+						return err
+					}
 				}
 			}
-			// No filter, collect everything.
-			if err := collectPachd(ctx, tw, pachClient, pachdContainerPrefix); err != nil {
-				return err
-			}
-			pipelineInfos, err := pachClient.ListPipeline(true)
-			if err != nil {
-				return err
-			}
-			for _, pipelineInfo := range pipelineInfos {
-				if err := s.handlePipelineRedirect(ctx, tw, pachClient, pipelineInfo, collectPipeline, collectWorker, redirect); err != nil {
-					return err
-				}
-			}
-			if wantAppLogs {
-				// All other pachyderm apps (console, pg-bouncer, etcd, etc.).
-				if err := s.appLogs(ctx, tw); err != nil {
-					return err
-				}
-			}
-			return nil
+
+			return s.collectEverything(pachClient, collectPachd, collectPipeline, collectWorker, redirect, wantAppLogs,
```

Comment:
I don't see a lot of value in delegating to another function here.  The list of arguments is almost longer than the actual code.

Do keep in mind that this function runs on pachd and every worker, so we can't do things like the database dump here without verifying that we're the first pachd and not a replica or a worker.